api/v1/securityintent_types.go

```diff
@@ -7,119 +7,50 @@ import (
-}
-
-// Resource defines the resources that the security policy applies to
-type SecurityIntentParams struct {
-	// Network: MatchProtocols
-	MatchProtocols []SecurityIntentMatchProtocol `json:"SecurityIntentMatchProtocols,omitempty"`
-
-	// Process: MatchPaths, MatchDirectories, MatchPatterns
-	// File: MatchPaths, MatchDirectories
-	MatchPaths       []SecurityIntentMatchPath      `json:"matchPaths,omitempty"`
-	MatchDirectories []SecurityIntentMatchDirectory `json:"matchDirectories,omitempty"`
-	MatchPatterns    []SecurityIntentMatchPattern   `json:"matchPatterns,omitempty"`
-
-	// Capabilities: MatchCapabilities
-	MatchCapabilities []SecurityIntentMatchCapability `json:"matchCapabilities,omitempty"`
-
-	// Syscalls: MatchSyscalls
-	MatchSyscalls     []SecurityIntentMatchSyscall     `json:"matchSyscalls,omitempty"`
-	MatchSyscallPaths []SecurityIntentMatchSyscallPath `json:"matchSyscallPaths,omitempty"`
-
-	FromCIDRSet []SecurityIntentCIDRSet `json:"fromCIDRSet,omitempty"`
-	ToPorts     []SecurityIntentToPort  `json:"toPorts,omitempty"`
-}
-
-// CIDRSet defines CIDR ranges for network policies
-type SecurityIntentCIDRSet struct {
-	CIDR string `json:"cidr,omitempty"`
-}
-
-// ToPort defines ports and protocols for network policies
-type SecurityIntentToPort struct {
-	Ports []SecurityIntentPort `json:"ports,omitempty"`
-}
-
-// Port defines a network port and its protocol
-type SecurityIntentPort struct {
-	Port     string `json:"port,omitempty"`
-	Protocol string `json:"protocol,omitempty"`
-}
-
-// SecurityIntentMatchProtocol defines a protocol for network policies
-type SecurityIntentMatchProtocol struct {
-	Protocol string `json:"protocol,omitempty"`
-}
-
-// MatchPath defines a path for process or file policies
-type SecurityIntentMatchPath struct {
-	Path string `json:"path,omitempty"`
-}
+	// ID is predefined in adapter ID pool.
+	// Used by security engines to generate corresponding security policies.
+	//+kubebuilder:validation:Pattern:="^[a-zA-Z0-9]*$"
+	ID string `json:"id"`
 
-// MatchDirectory defines a directory for process or file policies
-type SecurityIntentMatchDirectory struct {
-	Directory  string                     `json:"dir,omitempty"`
-	FromSource []SecurityIntentFromSource `json:"fromSource,omitempty"`
-}
+	// Description is human-readable explanation of the intent's purpose.
+	Description string `json:"description,omitempty"`
 
-// MatchPattern defines a pattern for process policies
-type SecurityIntentMatchPattern struct {
-	Pattern string `json:"pattern,omitempty"`
-}
+	// Action defines how the security policy will be enforced.
+	Action string `json:"action"`
 
-// MatchSyscall defines a syscall for syscall policies
-type SecurityIntentMatchSyscall struct {
-	Syscalls   []string            `json:"syscalls,omitempty"`
-	FromSource []SyscallFromSource `json:"fromSource,omitempty"`
-}
+	// Mode defines the enforcement behavior of the intent.
+	// Defaults to best-effort.
```

Comment:
Please mention "strict" here, and also what it means

Reply:
> Please mention "strict" here

We'll be updating our documentation to list all possible field values, so I don't think there's a need to mention `strict` here.

> what it means

 Mode defines the enforcement behavior of the intent.